substrate/rpc/src/chain/mod.rs

```diff
@@ -40,10 +40,11 @@ build_rpc_trait! {
 	/// Polkadot blockchain API
 	pub trait ChainApi {
 		type Metadata;
+		ASSOCIATED type Header;
```

Comment:
why caps?

Reply:
to distinguish from the rust-like grammar

Reply:
Ok, I might not understand how this works. What's really the difference between declaring `type Header` and `type Metadata`?

Reply:
`type Metadata;` already existed; it's some sort of marker that the macro uses to change the code that's generated.

i had to add the `ASSOCIATED` tag in order to allow for doc-comments of the type, since otherwise the grammar is ambiguous.